modules/account-lib/src/coin/algo/assetTransferBuilder.ts

```diff
@@ -71,10 +71,41 @@ export class AssetTransferBuilder extends TransferBuilder {
     );
   }
 
+  /**
+   * Computes the specific (sub) type for this Asset Transaction in algorand
+   * https://developer.algorand.org/docs/get-details/transactions/transactions/#asset-transfer-transaction
+   * @protected
+   */
   protected get transactionType(): TransactionType {
+    if (this.isOptOutAssetTxn()) {
+      return TransactionType.DisableToken;
+    }
+
+    if (this.isOptInAssetTrx()) {
+      return TransactionType.EnableToken;
+    }
+
     return TransactionType.Send;
   }
 
+  /**
+   * Check if this transaction to opt-in to an asset (allowlist)
+   * @see https://developer.algorand.org/articles/algos-asas/
+   * @private
+   */
+  private isOptInAssetTrx() {
+    return this._tokenId && !this.isOptOutAssetTxn() && BigInt(this._amount) === BigInt(0) && this._to === this._sender;
+  }
+
+  /**
+   * Check if this transaction to opt-out to an asset (do not allowlist)
+   * @see https://developer.algorand.org/articles/algos-asas/
+   * @private
+   */
+  private isOptOutAssetTxn() {
+    return this._tokenId && this._closeRemainderTo;
```

Comment:
Here also amount should be zero?

Reply:
it is not required to be zero, it seems you can send some amount to a destination account, send the remaining to the closeRemainderTo, and "close" this asset in the sender address.
https://developer.algorand.org/docs/get-details/transactions/transactions/#asset-transfer-transaction